R/plotScoresByCellOrLabel.R

```diff
@@ -0,0 +1,128 @@
+#' Plot SingleR scoring on a per label or per cell basis.
+#' @param results A \linkS4class{DataFrame} containing the output from \code{\link{SingleR}} or \code{\link{classifySingleR}}.
+#' @param cell.id Integer specifying which cell to show for \code{plotScoresSingleCell}
+#' @param labels.use String vector indicating what labels to show in \code{plotScoresSingleCell} and \code{plotScoresMultiLabel}
+#' If \code{labels.use} is left \code{NULL}, all labels available in \code{results} are presented.
+#' @param label String indicating which individual label to plot in \code{plotScoresSingleLabel}
+#' @param prune.calls Logical vector, the output of \code{\link{pruneScores}}.
+#' This input will be unnecessary once \code{\link{pruneScores}}'s output is added to the SingleR.results dataframe
+#' @param dots.on.top Logical which sets whether cell dots are plotted on top of, versus behind, the violin plots in \code{plotScoresSingleLabel} and \code{plotScoresMultiLabel}
+#' @param colors String vector that sets the colors.  Order of colors should be: `this label`, `this label - pruned`, `other label`, `other label - pruned`.
+#' @param size the size of the dots
+#' @name plotScoresByCellOrLabel
+#' @return Dot and violin plot of SingleR scores separated by whether this was the final call or not.
+#' If prune.calls are provided, will also separate and color based on whether cells' final calls were pruned versus not. 
+#'
+#' @details
+#' The \code{plotScoresSingleCell} function creates a jitter plot showing the scores of a single cell accross many labels.
+#' 
+#' The \code{plotScoresSingleLabel} function creates a jitter and violin plot showing the scores of all cells accross a single label.
+#' 
+#' The \code{plotScoresMultiLabel} function creates \code{plotScoresSingleLabel} plots for multiple individual labels. 
+#' 
+#' @author Daniel Bunis
+#' @examples
+#' example(SingleR, echo=FALSE)
+#' prune <- pruneScores(pred$scores)
+#' 
+#' plotScoresSingleCell(results = pred, prune.calls = prune, cell.id = 1)
+#' plotScoresSingleLabel(results = pred, prune.calls = prune, label = "B",
+#'     dots.on.top = TRUE)
+#' plotScoresMultiLabels(results = pred,
+#'     dots.on.top = TRUE, size = 0.5)
+#' 
+#' NULL
+NULL
+
+
+#' @describeIn plotScoresByCellOrLabel Plot scores accross labels of an individual cells
+#' @export
+plotScoresSingleCell <- function(results, cell.id, prune.calls = NULL,
+    labels.use = levels(as.factor(results$labels)), size = 2,
+    colors = c("#F0E442", "#56B4E9", "gray70", "gray40")){
+    if(length(colors)<4){stop("4 colors are expected.")}
+    if (is.null(names(colors))){names(colors) <- c('this label', 'this label - pruned', 'other label', 'other label - pruned')}
+    if (is.null(rownames(results))) {
+        rownames(results) <- seq_len(nrow(results))
+    }
+    df <- plotScores.dataGather(results, prune.calls, labels.use)
+    p <- ggplot(data = df[df$id == rownames(results)[cell.id],],
```

Comment:
You'll notice that I only `Suggests:` ggplot2. This is deliberate to avoid dragging in ggplot during installation and package loading. However, it means that all ggplot-related functions need `ggplot2::` in front of them.

Reply:
Noted!